pkg/shared/ruleset/disak8sstig/rules/274882_test.go

```diff
@@ -0,0 +1,275 @@
+			[]rule.CheckResult{{Status: rule.Errored, Message: "deployment: kube-apiserver does not contain container: kube-apiserver", Target: target}},
+			BeNil()),
+		Entry("should fail when encryption-provider-config is not set",
+			corev1.Container{Name: "kube-apiserver", Command: []string{}},
+			[]rule.CheckResult{{Status: rule.Failed, Message: "Option encryption-provider-config has not been set in the container command.", Target: target}},
+			BeNil()),
+	)
+
+	DescribeTable("Run cases for an EncryptionConfiguration object",
+		func(encryptionProviderConfig string, expectedCheckResult rule.CheckResult) {
+			ksDeployment.Spec.Template.Spec.Containers = []corev1.Container{
+				{
+					Name:    "kube-apiserver",
+					Command: []string{"./command"},
+					Args:    []string{"--encryption-provider-config=/etc/kubernetes/foo"},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "encryption-provider-config",
+							MountPath: "/etc/kubernetes/foo",
+						},
+					},
+				},
+			}
+
+			ksDeployment.Spec.Template.Spec.Volumes = []corev1.Volume{
+				{
+					Name: "encryption-provider-config",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "encryption-provider-config",
+							},
+						},
+					},
+				},
+			}
+			Expect(fakeClient.Create(ctx, ksDeployment)).To(Succeed())
+
+			configMap := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "encryption-provider-config",
+					Namespace: namespace,
+				},
+				Data: map[string]string{
+					"/etc/kubernetes/foo": encryptionProviderConfig,
+				},
+			}
+			Expect(fakeClient.Create(ctx, configMap)).To(Succeed())
+
+			r := &rules.Rule274882{Client: fakeClient, Namespace: namespace}
+			ruleResult, err := r.Run(ctx)
+
+			Expect(err).To(BeNil())
+			Expect(ruleResult.CheckResults[0]).To(Equal(expectedCheckResult))
+		},
+		Entry("should error when the encryption config is not valid yaml",
+			"foo",
+			rule.CheckResult{Status: rule.Errored, Message: "yaml: unmarshal errors:\n  line 1: cannot unmarshal !!str `foo` into apiserver.EncryptionConfiguration", Target: target},
+		),
+		Entry("should fail when the encryption config has no resources configured",
+			`
+apiVersion: apiserver.config.k8s.io/v1
+kind: EncryptionConfiguration`,
+			rule.CheckResult{Status: rule.Failed, Message: "Secrets are not explicitly encrypted at REST.", Target: target},
+		),
+		Entry("should fail when the encryption config does not include secrets as a resource for encryption",
+			`
+apiVersion: apiserver.config.k8s.io/v1
+kind: EncryptionConfiguration
+resources:
+- resources:
+  - pods
+  providers:
+  - aesgcm:
+    keys:
+    - name: key1
+      secret: c2VjcmV0IGlzIHNlY3VyZQ==`,
+			rule.CheckResult{Status: rule.Failed, Message: "Secrets are not explicitly encrypted at REST.", Target: target},
```

Comment:
```suggestion
  providers:
  - aesgcm:
      keys:
      - name: key1
        secret: c2VjcmV0IGlzIHNlY3VyZQ==`,
			rule.CheckResult{Status: rule.Failed, Message: "Secrets are not explicitly encrypted at REST.", Target: target},
```

Please fix the other occurrences as well.